scripts/video_ir.py

```diff
@@ -89,10 +92,73 @@ def __str__(self) -> str:
 
       return result
 
+class Verse:
+
+    def __init__(self, verse_data) -> None:
+        self.count = verse_data.get("count")
+        self.transliteration = verse_data.get("verse_trans")
+        self.sanskrit = verse_data.get("verse_sanskrit")
+        self.meaning = verse_data.get("verse_meaning")
+
+        return
+
+    def get_similarity_score_for_text(self, other_text):
+        if not other_text:
+            return
+
+        return instance.compare(self.sanskrit, other_text)
+
+
+    def __repr__(self) -> str:
+        msg = f"{[self.count]}\n{self.sanskrit}\n"
+        return msg
+
+class ScrapedText:
+    def __init__(self, file_data) -> None:
+        self.title = file_data.get("title")
+        self.description = file_data.get("description")
+        self.verses = [Verse(verse_data) for verse_data in file_data.get("verses")]
+
+        return
+
 def main():
-    data = read_file(sys.argv[1] or "chalisa.json")
-    captions = VideoCaptions(data)
-    captions.create_srt_file(sys.argv[2] or "chalisa.srt")
+    caption_data = read_file(sys.argv[1] if len(sys.argv) - 1 > 0 else "chalisa.json")
+    captions = VideoCaptions(caption_data)
+    # captions.create_srt_file(sys.argv[2] if len(sys.argv) - 1 > 1 else "chalisa.srt")
+
+    scraped_data = read_file(sys.argv[1] if len(sys.argv) - 1 > 0 else "chalisa_scraped.json")
+    scraped_text = ScrapedText(scraped_data)
+
+    test_input_verse = scraped_text.verses[4]
+    test_input_captions = captions.caption_events[45].event_text
+    all_captions = [event.event_text for event in captions.caption_events]
+    # print(">>> test input verse:\n", test_input_verse)
+    # print("test_input_captions\n", test_input_captions)
+    # print("sim:", test_input_verse.get_similarity_score_for_text(test_input_captions))
+
+    for verse_idx, verse in enumerate(scraped_text.verses):
+        scores = []
+        for caption_idx, caption in enumerate(all_captions):
+            score = verse.get_similarity_score_for_text(caption)
+
+            if (score == 1):
+                print(f">> DIRECT MATCH: \n verse_idx {verse_idx} matching caption idx: {caption_idx}")
+                # print(f">> MATCH: \ntest input: {test_input_verse}\n matching: {caption}")
+            scores.append(score or -1)
+
+        highest_score = max(scores)
+        most_likely_caption_idx = highest_score_idx = scores.index(highest_score)
+        print(f"Verse idx {verse_idx} --> likely to be --> {most_likely_caption_idx} with score of {highest_score}")
```

Comment:
output dump (might be buggy, will double check in the morning) 

```
ॐ ❯ ./video_ir.py
>> Reading chalisa.json as a json file
>> Reading chalisa_scraped.json as a json file
Verse idx 0 --> likely to be --> 0 with score of -1
>> DIRECT MATCH: 
 verse_idx 1 matching caption idx: 13
Verse idx 1 --> likely to be --> 13 with score of 1
Verse idx 2 --> likely to be --> 0 with score of -1
>> DIRECT MATCH: 
 verse_idx 3 matching caption idx: 35
Verse idx 3 --> likely to be --> 35 with score of 1
>> DIRECT MATCH: 
 verse_idx 4 matching caption idx: 45
Verse idx 4 --> likely to be --> 45 with score of 1
Verse idx 5 --> likely to be --> 0 with score of -1
Verse idx 6 --> likely to be --> 0 with score of -1
Verse idx 7 --> likely to be --> 0 with score of -1
Verse idx 8 --> likely to be --> 0 with score of -1
Verse idx 9 --> likely to be --> 0 with score of -1
>> DIRECT MATCH: 
 verse_idx 10 matching caption idx: 75
Verse idx 10 --> likely to be --> 75 with score of 1
Verse idx 11 --> likely to be --> 0 with score of -1
Verse idx 12 --> likely to be --> 0 with score of -1
Verse idx 13 --> likely to be --> 0 with score of -1
>> DIRECT MATCH: 
 verse_idx 14 matching caption idx: 109
Verse idx 14 --> likely to be --> 109 with score of 1
Verse idx 15 --> likely to be --> 0 with score of -1
Verse idx 16 --> likely to be --> 0 with score of -1
Verse idx 17 --> likely to be --> 0 with score of -1
Verse idx 18 --> likely to be --> 0 with score of -1
Verse idx 19 --> likely to be --> 0 with score of -1
>> DIRECT MATCH: 
 verse_idx 20 matching caption idx: 143
Verse idx 20 --> likely to be --> 143 with score of 1
Verse idx 21 --> likely to be --> 0 with score of -1
Verse idx 22 --> likely to be --> 0 with score of -1
>> DIRECT MATCH: 
 verse_idx 23 matching caption idx: 163
Verse idx 23 --> likely to be --> 163 with score of 1
Verse idx 24 --> likely to be --> 0 with score of -1
Verse idx 25 --> likely to be --> 0 with score of -1
Verse idx 26 --> likely to be --> 0 with score of -1
Verse idx 27 --> likely to be --> 0 with score of -1
Verse idx 28 --> likely to be --> 0 with score of -1
Verse idx 29 --> likely to be --> 0 with score of -1
Verse idx 30 --> likely to be --> 0 with score of -1
Verse idx 31 --> likely to be --> 0 with score of -1
Verse idx 32 --> likely to be --> 0 with score of -1
Verse idx 33 --> likely to be --> 0 with score of -1
Verse idx 34 --> likely to be --> 0 with score of -1
Verse idx 35 --> likely to be --> 0 with score of -1
Verse idx 36 --> likely to be --> 0 with score of -1
Verse idx 37 --> likely to be --> 0 with score of -1
Verse idx 38 --> likely to be --> 0 with score of -1
Verse idx 39 --> likely to be --> 0 with score of -1
Verse idx 40 --> likely to be --> 0 with score of -1
>> DIRECT MATCH: 
 verse_idx 41 matching caption idx: 267
Verse idx 41 --> likely to be --> 267 with score of 1
Verse idx 42 --> likely to be --> 0 with score of -1
~/Projects/vyasa/scripts ritesh/operation/video-IR *11 ?9                                  L 1.65 RAM 6G SWP 0B █ 90% 22:49:22
ॐ
```

Reply:
i think maybe it's just a matter of how I'm calling the function, if I use the tool on [here](https://libindic.org/Similar%20Texts), then sometimes I do get non -1 answers indicating that the function should work...

Reply:
## slightly better but still dismal results 

approach: trying to map caption events (small text strings) to longer string (verse strings) gives the following: 

```bash
ॐ ❯ ./video_ir.py
>> Reading chalisa.json as a json file
>> Reading chalisa_scraped.json as a json file
C 14 V 2
        score: 1
caption:         बुद्धिहीन  तनु  जानिके
verse:   बुद्धिहीन तनु जानिके सुमिरौं पवन-कुमार ।

बल बुधि बिद्या देहु मोहिं हरहु कलेस बिकार ॥
C 36 V 4
        score: 1
caption:         रामदूत  अतुलित
verse:   राम दूत अतुलित बल धामा ।

अञ्जनि-पुत्र पवनसुत नामा ॥२॥
C 46 V 5
        score: 1
caption:         महाबीर  बिक्रम  बजरंगी
verse:   महाबीर बिक्रम बजरङ्गी ।

कुमति निवार सुमति के सङ्गी ॥३॥
C 76 V 11
        score: 1
caption:         सूक्ष्म  रूप
verse:   सूक्ष्म रूप धरि सियहिं दिखावा ।

बिकट रूप धरि लङ्क जरावा ॥९॥
C 110 V 15
        score: 1
caption:         सहस  बदन  तुम्हरो  तो  जज  का  या  वेब  अस  कहि
verse:   सहस बदन तुह्मारो जस गावैं ।

अस कहि श्रीपति कण्ठ लगावैं ॥१३॥
C 144 V 21
        score: 1
caption:         प्रभु  मुद्रिका
verse:   प्रभु मुद्रिका मेलि मुख माहीं ।

जलधि लाँघि गये अचरज नाहीं ॥१९॥
C 164 V 24
        score: 1
caption:         सब  सुख  लहै  तुम्हारी
verse:   सब सुख लहै तुह्मारी सरना ।

तुम रच्छक काहू को डर ना ॥२२॥
C 268 V 42
        score: 1
caption:         तुलसीदास  सदा  हरि  चेरा
verse:   तुलसीदास सदा हरि चेरा ।

कीजै नाथ हृदय महँ डेरा ॥४०॥
```